service/dap/server.go

```diff
@@ -811,28 +831,34 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 
 	// Prepare the debug executable filename, build flags and build it
 	if mode == "debug" || mode == "test" {
-		output := args.Output
-		if output == "" {
-			output = defaultDebugBinary
-		}
-		output = cleanExeName(output)
-		debugbinary, err := filepath.Abs(output)
-		if err != nil {
-			s.sendInternalErrorResponse(request.Seq, err.Error())
-			return
+		debugbinary := args.Output
+		if debugbinary == "" {
+			debugbinary = s.tempDebugBinary()
 		}
 
-		buildFlags := args.BuildFlags
+		if !filepath.IsAbs(debugbinary) {
+			o, err := filepath.Abs(debugbinary)
+			if err != nil {
+				s.sendInternalErrorResponse(request.Seq, err.Error())
```

Comment:
this should `s.sendErrorResponse(request.Request, FailedToLaunch, "Failed to launch", err.Error())`

Reply:
Why is it user's error? (previously it was internal error).

Reply:
Previous code was wrong. Failure to launch is not a user error. It's an error to satisfy the launch request for any reason (bad user input, system malfunction, etc).  The internal errors were add for cases where we could not associate the error with the specific request (when not being able to parse the command or panic'ing).

Reply:
Done.